### DISCUSSION TITLE
Comparative/Superlative Tags in German

### DISCUSSION THREAD

**Discussion by frahen** [explosion/spaCy#12789] [Help: Coding & Implementations] [answered]
Hey,

I'm new to using spacy, so apologies if my question is silly. For my work, I need to update a script that analyzes a text, which can be German or English, for comparative and superlative, among other things. As language models we use en_core_web_md and de_core_news_md. For English, a matcher is implemented that searches for the tags JJS, JJR, etc. that directly mark comparative or superlative words. For German I can't find such tags. Are there other tags or so to find comparative/superlative? If not, can you give me some advice on how to proceed? Thank you!

**Comment by victorialslocum** [comment, marked as answer]
Hi @frahen,

Thank you for your question! We’re always happy to help, and this isn’t a 100% obvious answer.

You can start by checking the https://spacy.io/models/de page under ‘label scheme’ on whether the labels you're looking for are there. You might want to check the morphologizer labels as well. If you have questions on what any of the labels mean, you can use `print(spacy.explain("..."))`, though I’m not sure it works for morph labels/feats.

However, that morphologizer list may be truncated, and also maybe hard to find what you’re looking for exactly. If it’s not in the fine-grained tags, you can look at the general UD morphologizer features here: https://universaldependencies.org/u/feat/all.html. You can also look at the data in `nlp.meta` (https://spacy.io/api/language#meta) for the morphologizer rather than the truncated lists in https://spacy.io/models/de to see what’s available/annotated for the language.

**Comment by frahen** [comment]
Great! Thanks @victorialslocum, that helped me a lot.

In the morphologizer features, I found superlative and comparative in the degree inflection feature. Can I use them in a matcher rule or how can I use them?

**Comment by victorialslocum** [comment]
Glad it helped! 

Yes definitely! You can find more information on available pattern attributes here: https://spacy.io/usage/rule-based-matching#adding-patterns-attributes. The `MORPH` tag should do the trick. The [section on extended syntax](https://spacy.io/usage/rule-based-matching#adding-patterns-attributes-extended) might help too.

**Comment by frahen** [comment]
A little update. After experimenting a bit with the Matcher rules, I finally found rules that recognize comparative and superlative in German texts. Not every case can be found, but it has a high accuracy for that, which is more important in my case. Thanks again, it really helped me :)

**Reply by EvaDavita** [reply]
Hi @frahen , I happen to have the exact same task, to recognize comparative and superlative German words in texts. Would you mind sharing the code that worked for you? Would be a really big help! :)

**Reply by frahen** [reply]
Hey @EvaDavita, sure! In the end I used the following 2 matcher rules:

```
comparative_rule_de = [[{"MORPH": {"IS_SUPERSET": ["Degree=Cmp"]}}]]
superlative_rule_de = [[{"MORPH": {"IS_SUPERSET": ["Degree=Sup"]}}]]
```

They didn't detect all cases, but at least in my tests it never found superlatives or comperatives where there were none. In my use case this was more important. Hope this helps! Feel free to ask If you have more questions. :)

**Reply by EvaDavita** [reply]
Thank you so much @frahen ! Much appreciated. Sounds like that’s exactly what I need!
